docs/architecture/adr-030-consensus-refactor.md

```diff
@@ -126,6 +126,184 @@ func TestConsensusXXX(t *testing.T) {
 }
 ```
 
+
+## Consensus Executor
+
+## Consensus Core
+
+```go
+type Event interface{}
+
+type EventNewHeight struct {
+    Height           int64
+    ValidatorSetSize int
+    ValidatorId      int
+}
+
+type EventNewRound struct {
+    Height           int64
+    Round            int
+}
+
+type EventProposal struct {
+    Height           int64
+    Round            int
+    Timestamp        Time
+    BlockID          BlockID
+    POLRound         int
+    Sender           int   
+}
+
+type Majority23PrevotesBlock struct {
+    Height           int64
+    Round            int
+    BlockID          BlockID
+}
+
+type Majority23PrecommitBlock struct {
+    Height           int64
+    Round            int
+    BlockID          BlockID
+}
+
+type Majority23PrevotesAny struct {
+    Height           int64
+    Round            int
+}
+
+type Majority23PrecommitAny struct {
+    Height           int64
+    Round            int
+}
+
+type TimeoutPropose struct {
```

Comment:
We should just define a struct with `Height` and `Round` once and let all these alias it, eg. `type TimeoutPropose HeightRoundMsg`